### PR TITLE
[wren] update to v0.4.0

### LINKED ISSUE #24316: [wren] update to 0.4.0
Library name: wren

New version number: 0.4.0

https://github.com/wren-lang/wren/releases/tag/0.4.0

### PR DESCRIPTION
**Pull request by Adela0814**
Fix #24316

no feature needs to test

### CODE REVIEW

File: ports/wren/portfile.cmake [hunk 1]
```diff
@@ -3,25 +3,24 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
...
     OUT_SOURCE_PATH SOURCE_PATH
     REPO wren-lang/wren
-    REF d1a0d0682ac072fa20f2dcca356dac06565e93a1
-    SHA512 d6b7555da10ea209d04308467563ec401f699a22ca8524e4ae65fd050b0403c5e70430047dbb6a883b8021ab50525c17bb63d76543e7c59dfe4b521cfde6c413
+    REF 4a18fc489f9ea3d253b20dd40f4cdad0d6bb40eb #0.4.0
+    SHA512 b3d79e9cb647e52db06d5cddfc8c93d05ae0e8d87f0f879ac2b812fcc7f55e018d21d3b04d62eaeb12e6d931b5e43fbe357b187e7f446e86e39be015c51c2eee
     HEAD_REF master
 )
 
 file(COPY ${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt DESTINATION ${SOURCE_PATH})
```
Comment (LilyWangLL) [inline ports/wren/portfile.cmake:11]:
```suggestion
file(COPY "${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt" DESTINATION "${SOURCE_PATH}")

```

File: ports/wren/portfile.cmake [hunk 2]
```diff
@@ -3,25 +3,24 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
...
 
-vcpkg_configure_cmake(
+vcpkg_cmake_configure(
     SOURCE_PATH ${SOURCE_PATH}
```
Comment (LilyWangLL) [inline ports/wren/portfile.cmake:14]:
```suggestion
    SOURCE_PATH "${SOURCE_PATH}"

```

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

File: versions/baseline.json [hunk 3]
```diff
@@ -7497,8 +7497,8 @@
       "port-version": 2
     },
     "wren": {
-      "baseline": "2019-07-01",
-      "port-version": 1
+      "baseline": "0.4.0",
+      "port-version": 0
```
Comment (LilyWangLL) [inline versions/baseline.json:7501]:
```suggestion
```
Update to new version need remove `"port-version"`.

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

**Review by LilyWangLL — APPROVED** [review]

**Comment by strega-nil-ms** [comment]
Thanks!

**Pull request merged**